vortex-api/src/main/java/mil/army/usace/hec/vortex/temporal/RecordIndexQuery.java

```diff
@@ -0,0 +1,20 @@
+package mil.army.usace.hec.vortex.temporal;
+
+import mil.army.usace.hec.vortex.VortexDataType;
+
+import java.time.ZonedDateTime;
+import java.util.List;
+
+interface RecordIndexQuery {
+    List<Integer> query(ZonedDateTime startTime, ZonedDateTime endTime);
+    ZonedDateTime getEarliestStartTime();
+    ZonedDateTime getLatestEndTime();
+
+    static RecordIndexQuery of(VortexDataType dataType, List<VortexTimeRecord> indexedRecords) {
```

Comment:
I usually pass args as an array so that type can be enforced.

Reply:
https://github.com/HydrologicEngineeringCenter/Vortex/pull/116#discussion_r1791894175